crates/tabby-scheduler/src/crawl/mod.rs

```diff
@@ -0,0 +1,97 @@
+mod types;
+
+use std::process::Stdio;
+
+use async_stream::stream;
+use futures::{Stream, StreamExt};
+use readable_readability::Readability;
+use tokio::io::AsyncBufReadExt;
+use tracing::warn;
+use url::Url;
+
+use self::types::{CrawledDocument, KatanaRequestResponse};
+
+async fn crawl_url(start_url: &str) -> impl Stream<Item = KatanaRequestResponse> {
+    let mut child = tokio::process::Command::new("katana")
```

Comment:
Can't use something like `spider` crate?

Reply:
It's still in the prototyping phase—primarily because I don't want to limit such a complex component to a monolithic Rust stack. However, I might change my mind in later stages, depending on whether we have sufficient third-party library support.